### PR TITLE
[3] Investment Page Updates

### PR DESCRIPTION
**Pull request by aivantg**
[//]: # "These comments meant for your reference, they are invisible and don't need to be deleted"

## What's new in this PR?

[//]: # '#####################################'
[//]: # '####### YOUR ANSWER BELOW ###########'

- Finished up Payment functionality for buy shares page to work with #121  updates

[//]: # '############## END ##################'

### How to Review

[//]: # 'The order in which to review files and 
what to expect when testing locally'
[//]: # '#####################################'
[//]: # '####### YOUR ANSWER BELOW ###########'
[//]: # '############## END ##################'

### Related PRs

[//]: # "Optional - related PRs you're waiting on
/ PRs that will conflict, etc"
[//]: # '#####################################'
[//]: # '####### YOUR ANSWER BELOW ###########'

Dependent on #121 

[//]: # '############## END ##################'

### Next steps

[//]: # "What doesn't work yet, what's NOT in this 
PR/has to be done "
[//]: # '#####################################'
[//]: # '####### YOUR ANSWER BELOW ###########'
[//]: # '############## END ##################'

### Tests Performed, Edge Cases

[//]: # '#####################################'
[//]: # '####### YOUR ANSWER BELOW ###########'
[//]: # '############## END ##################'

### Screenshots

[//]: # '#####################################'
[//]: # '#### YOUR SCREENSHOTS BELOW ########'
[//]: # '############## END ##################'

CC: @dfangshuo

### CODE REVIEW

File: src/lib/paypalUtils.js [hunk 1]
```diff
@@ -22,9 +22,10 @@ const constructPaymentRecord = (details, data, ownerId) => {
   };
 };
 
-const recordSharePayment = async (details, data, ownerId) => {
+const recordSharePayment = async (details, data, ownerId, numberOfShares) => {
   const payment = constructPaymentRecord(details, data, ownerId);
   await createPayment({ ...payment, type: BUY_SHARES_TYPE });
+  await updateOwner(ownerId, { numberOfShares });
```
Comment (dfangshuo) [inline src/lib/paypalUtils.js:28]: we should have a check here that makes sure the new numberOfShares >= 0 and >= the previous number of shares. This can be an assertion or a log statement etc.

File: src/screens/general/Investment.js [hunk 2]
```diff
@@ -71,9 +71,12 @@ class Investment extends React.PureComponent {
...
                 <div className="investments-buttons">
-                  <div className="investments-buy-shares-button">
-                    <Link to="/buyshares">Buy Shares</Link>
-                  </div>
+                  {owner.numberOfShares !== 10 && (
+                    <div className="investments-buy-shares-button">
+                      <Link to="/buyshares">Buy Shares</Link>
+                    </div>
+                  )}
+
```
Comment (dfangshuo) [inline src/screens/general/Investment.js:79]: maybe have this be grayed out instead of making the button disappear altogether

File: src/screens/general/BuyShares.js [hunk 3]
```diff
@@ -13,10 +17,25 @@ class BuyShares extends React.PureComponent {
...
+    const { sharesBuying } = this.state;
+    this.setState({ loading: true });
+    await recordSharePayment(
+      details,
+      data,
+      owner.id,
+      owner.numberOfShares + sharesBuying
+    );
+    await refreshUserData(owner.id);
+    history.push(Constants.HOME_ROUTE);
```
Comment (dfangshuo) [inline src/screens/general/BuyShares.js:36]:
```suggestion
    history.push(Constants.HOME_ROUTE); // TODO: add redirect to payment confirmation screen
```

**Comment by dfangshuo** [comment]
remember to merge [4], [3], [2], [1] to avoid weird bugs!!

File: src/lib/paypalUtils.js (hunk 1 above)
Comment (aivantg) [reply src/lib/paypalUtils.js:28]: not a bad idea. this logic currently happens on the frontend but might not be bad to validate it here too

File: src/screens/general/Investment.js (hunk 2 above)
Comment (aivantg) [reply src/screens/general/Investment.js:79]: going to defer to iris for a proper disabled button rather than hacking one together for now
